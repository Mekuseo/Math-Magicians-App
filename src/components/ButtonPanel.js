/* eslint-disable react/destructuring-assignment */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import Button from './Button';

function ButtonPanel(props) {
  const handleClick = (buttonName) => {
    props.clickHandler(buttonName);
  };

  return (
    <div className="component-button-panel">
      <div className="component-button-row">
        <Button name="AC" clickHandler={handleClick} />
        <Button name="+/-" clickHandler={handleClick} />
        <Button name="%" clickHandler={handleClick} />
        <Button name="/" clickHandler={handleClick} orange />
      </div>
      <div className="component-button-row">
        <Button name="7" clickHandler={handleClick} />
        <Button name="8" clickHandler={handleClick} />
        <Button name="9" clickHandler={handleClick} />
        <Button name="*" clickHandler={handleClick} orange />
      </div>
      <div className="component-button-row">
        <Button name="4" clickHandler={handleClick} />
        <Button name="5" clickHandler={handleClick} />
        <Button name="6" clickHandler={handleClick} />
        <Button name="-" clickHandler={handleClick} orange />
      </div>
      <div className="component-button-row">
        <Button name="1" clickHandler={handleClick} />
        <Button name="2" clickHandler={handleClick} />
        <Button name="3" clickHandler={handleClick} />
        <Button name="+" clickHandler={handleClick} orange />
      </div>
      <div className="component-button-row">
        <Button name="0" clickHandler={handleClick} wide />
        <Button name="." clickHandler={handleClick} />
        <Button name="=" clickHandler={handleClick} orange />
      </div>
    </div>
  );
}

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
