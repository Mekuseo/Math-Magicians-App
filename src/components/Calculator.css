.component-app {
  /* max-width: 260px; */
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: #eee;
  border-radius: 10px;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);
}

.component-app > * {
  padding: 1rem;
}

.calculator-page {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.calculator-container {
  display: flex;
  flex: 1;
  gap: 300px;
}

.calculator-container h1 {
  flex: 0 0 25%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 2rem;
}

.calculator-display {
  /* flex: 1; */
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: #eee;
  border-radius: 10px;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);
  padding: 1rem;
}

@media (min-width: 768px) {
  .calculator-container h1 {
    font-size: 2rem;
    margin-left: 30px;
  }
}

@media (min-width: 768px) {
  .component-app {
    max-width: none;
  }
}
