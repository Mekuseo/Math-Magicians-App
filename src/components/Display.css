.display {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  font-size: 3rem;
  padding: 0 1rem;
  background-color: #777171;
  color: #fff;
  height: 6rem;
  box-shadow: inset 0 0 5px rgba(0,0,0,0.2);
}
