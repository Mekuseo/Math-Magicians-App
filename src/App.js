/* eslint-disable no-unused-vars */
import React from 'react';
import Calculator from './components/Calculator';

class App extends React.Component {
  constructor(props) {
    super();
  }

  render() {
    return (
      <Calculator />
    );
  }
}

export default App;
