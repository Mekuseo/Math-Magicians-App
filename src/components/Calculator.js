import React, { useState } from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from './logic/Calculate';
import NavBar from './pages/NavBar/NavBar';
import HomePage from './pages/HomePage/HomePage';
import QuotesLogic from './pages/Quotes/QuotesLogic';
import './Calculator.css';

function Calculator() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setState((prevState) => calculate(prevState, buttonName));
  };

  const { total, next } = state;
  const displayValue = next || total || '0';

  return (
    <Router>
      <div className="calculator-page">
        <NavBar />
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/home" element={<HomePage />} />
          <Route
            path="/calculator"
            element={(
              <div className="calculator-container">
                <h1>Let us do Some Maths</h1>
                <div className="calculator-display">
                  <Display value={displayValue} data-testid="display" />
                  <ButtonPanel clickHandler={handleClick} />
                </div>
              </div>
            )}
          />
          <Route path="/quote" element={<QuotesLogic />} />
        </Routes>
      </div>
    </Router>
  );
}

export default Calculator;
