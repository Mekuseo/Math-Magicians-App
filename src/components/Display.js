import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import './Display.css';

function Display({ value }) {
  return <div className="display">{value}</div>;
}

Display.propTypes = {
  value: PropTypes.string.isRequired,
};

export default Display;
