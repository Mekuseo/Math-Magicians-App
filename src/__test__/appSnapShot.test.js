import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import ButtonPanel from '../components/ButtonPanel';
import Display from '../components/Display';
import App from '../App';

it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const clickHandler = jest.fn();
  const tree = renderer
    .create(<ButtonPanel clickHandler={clickHandler} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Display>123</Display>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
