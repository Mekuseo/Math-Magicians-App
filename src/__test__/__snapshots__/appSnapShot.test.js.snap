// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator-page"
>
  <div
    className="navbar"
  >
    <h1
      className="nav-header"
    >
      Math Magicians
    </h1>
    <ul
      className="nav-list"
    >
      <li
        className="border-line"
      >
        <a
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="border-line"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <div
    className="home-page"
  >
    <h1
      className="header"
    >
      Math Magicians
    </h1>
    <p
      className="description"
    >
      Math Magicians is a mobile application designed to help users improve their mathematical skills through a fun and engaging gaming experience. The app features a variety of math games that challenge users to solve problems quickly and accurately, with increasing difficulty as they progress. The games include basic arithmetic operations such as addition, subtraction, multiplication, and division, as well as more advanced concepts such as algebra and geometry. The app offers a range of game modes, including timed challenges, puzzle-based games, and multiplayer options, allowing users to compete against friends and other players from around the world. One of the key features of Math Magicians is its adaptive learning algorithm, which adjusts the difficulty of the games based on the user's performance. This ensures that the user is always challenged, but not overwhelmed, as they progress through the app. The app also includes a variety of customization options, allowing users to personalize their experience with different themes, backgrounds, and avatars. Users can also track their progress and performance through detailed analytics and performance reports. Overall, Math Magicians is a fun and effective way to improve your mathematical skills, whether you're a student looking to improve your grades, an adult seeking to sharpen your mental math skills, or simply someone who enjoys a good challenge.
    </p>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="calculator-container"
>
  <div
    className="component-button-panel"
  >
    <div
      className="component-button-row"
    >
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="component-button orange"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
    </div>
    <div
      className="component-button-row"
    >
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="component-button orange"
        onClick={[Function]}
        type="button"
      >
        *
      </button>
    </div>
    <div
      className="component-button-row"
    >
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="component-button orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="component-button-row"
    >
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="component-button orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="component-button-row"
    >
      <button
        className="component-button  wide"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="component-button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="component-button orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<div
  className="display"
/>
`;

exports[`renders correctly 4`] = `
<div
  className="calculator-page"
>
  <div
    className="navbar"
  >
    <h1
      className="nav-header"
    >
      Math Magicians
    </h1>
    <ul
      className="nav-list"
    >
      <li
        className="border-line"
      >
        <a
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="border-line"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <div
    className="home-page"
  >
    <h1
      className="header"
    >
      Math Magicians
    </h1>
    <p
      className="description"
    >
      Math Magicians is a mobile application designed to help users improve their mathematical skills through a fun and engaging gaming experience. The app features a variety of math games that challenge users to solve problems quickly and accurately, with increasing difficulty as they progress. The games include basic arithmetic operations such as addition, subtraction, multiplication, and division, as well as more advanced concepts such as algebra and geometry. The app offers a range of game modes, including timed challenges, puzzle-based games, and multiplayer options, allowing users to compete against friends and other players from around the world. One of the key features of Math Magicians is its adaptive learning algorithm, which adjusts the difficulty of the games based on the user's performance. This ensures that the user is always challenged, but not overwhelmed, as they progress through the app. The app also includes a variety of customization options, allowing users to personalize their experience with different themes, backgrounds, and avatars. Users can also track their progress and performance through detailed analytics and performance reports. Overall, Math Magicians is a fun and effective way to improve your mathematical skills, whether you're a student looking to improve your grades, an adult seeking to sharpen your mental math skills, or simply someone who enjoys a good challenge.
    </p>
  </div>
</div>
`;
